core/lib/tests/file_server.rs

```diff
@@ -39,29 +46,44 @@ static INDEXED_DIRECTORIES: &[&str] = &[
     "inner/",
 ];
 
-fn assert_file(client: &Client, prefix: &str, path: &str, exists: bool) {
+fn assert_file(client: &Client, prefix: &str, path: &str, exists: bool, compressed: bool) {
     let full_path = format!("/{}/{}", prefix, path);
-    let response = client.get(full_path).dispatch();
+    let mut request = client.get(full_path);
+    request.add_header(Header::new("Accept-Encoding", "gzip"));
+    let mut response = request.dispatch();
     if exists {
         assert_eq!(response.status(), Status::Ok);
 
-        let mut path = static_root().join(path);
+        let mut path = match compressed {
+            true  => static_root().join(format!("{path}.gz")),
+            false => static_root().join(path),
+        };
         if path.is_dir() {
             path = path.join("index.html");
         }
 
         let mut file = File::open(path).expect("open file");
-        let mut expected_contents = String::new();
-        file.read_to_string(&mut expected_contents).expect("read file");
-        assert_eq!(response.into_string(), Some(expected_contents));
+        let mut expected_contents = vec![];
+        file.read_to_end(&mut expected_contents).expect("read file");
+
+        let mut actual = vec![];
+        response.read_to_end(&mut actual).expect("read response");
+
+        let ce: Vec<&str> = response.headers().get("Content-Encoding").collect();
```

Comment:
Would it make sense to check the Content-Type header similar to how you have here, and have a pre-defined header for GZipped files, and then other stuff otherwise?

Reply:
Non zipped files would not have a [content-encoding header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding) and I was only implementing gzip, so expanded values would need to be checked in the future if adding support for other compression schemes.